packages/gatsby/src/state-machines/waiting/index.ts

```diff
@@ -34,12 +34,33 @@ export const waitingStates: MachineConfig<IWaitingContext, any, any> = {
         },
         // We only listen for this when idling because if we receive it at any
         // other point we're already going to create pages etc
-        QUERY_FILE_CHANGED: {
+        SOURCE_FILE_CHANGED: {
+          target: `aggregatingFileChanges`,
+        },
+      },
+    },
+    aggregatingFileChanges: {
+      // Sigh. This is because webpack doesn't expose the Watchpack
+      // aggregated file invalidation events. If we compile immediately,
+      // we won't pick up the changed files
+      after: {
+        // The aggregation timeout
+        200: {
```

Comment:
How did you choose this number? Is this internal webpack's timeout?

Reply:
Yeah, this is the default value that webpack passes to watchpack.